File: week 3 - Greedy Algorithms/1_money_change/change.py
# Uses python3
# import numpy as np
import sys

def get_change(m):
    coins= int(m/10)
    tens=m%10
    fives=tens%5
    if tens==0:
        return coins
    elif fives==0:
        return coins+int(tens/5)
    else:
        return coins+int(tens/5)+fives


if __name__ == '__main__':
    # rand_num=np.random.randint(1,1000)
    m = int(sys.stdin.read())
    print(get_change(m))

File: week 3 - Greedy Algorithms/1_money_change/change.cpp
#include <iostream>

int get_change(int m) {
  int tens,fives,coins;
  coins= m/10;
  tens=m%10;
  fives=tens%5;
  if (tens==0)
      return coins;
  else if (fives==0)
      return coins+int(tens/5);
  else
      return coins+int(tens/5)+fives;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
}
